app/controllers/api/proxy.php

```diff
@@ -0,0 +1,296 @@
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_PROXY_RULE)
+    ->param('domain', null, new DomainValidator(), 'Domain name.')
+    ->param('resourceType', null, new WhiteList(['api', 'function']), 'Action definition for the rule. Possible values are "api", "function", or "redirect"')
+    ->param('resourceId', '', new UID(), 'ID of resource for the action type. If resourceType is "api" or "url", leave empty. If resourceType is "function", provide ID of the function.', true)
+    ->inject('response')
+    ->inject('project')
+    ->inject('events')
+    ->inject('dbForConsole')
+    ->inject('dbForProject')
+    ->action(function (string $domain, string $resourceType, string $resourceId, Response $response, Document $project, Event $events, Database $dbForConsole, Database $dbForProject) {
+        $document = $dbForConsole->findOne('rules', [
+            Query::equal('domain', [$domain]),
+        ]);
+
+        if ($document && !$document->isEmpty()) {
+            if ($document->getAttribute('projectId') === $project->getId()) {
+                $resourceType = $document->getAttribute('resourceType');
+                $resourceId = $document->getAttribute('resourceId');
+                $message = "Domain already assigned to '{$resourceType}' service";
+                if (!empty($resourceId)) {
+                    $message .= " with ID '{$resourceId}'";
+                }
+
+                $message .= '.';
+            } else {
+                $message = "Domain already assigned to different project.";
+            }
+
+            throw new Exception(Exception::RULE_ALREADY_EXISTS, $message);
+        }
+
+        $resourceInternalId = '';
+
+        if ($resourceType == 'function') {
+            if (empty($resourceId)) {
+                throw new Exception(Exception::RULE_RESOURCE_ID_MISSING);
+            }
+
+            $function = $dbForProject->getDocument('functions', $resourceId);
+
+            if ($function->isEmpty()) {
+                throw new Exception(Exception::RULE_RESOURCE_ID_NOT_FOUND);
+            }
+
+            $resourceInternalId = $function->getInternalId();
+        }
+
+        $domain = new Domain($domain);
+
+        $ruleId = ID::unique();
+
+        $status = 'created';
+        $functionsDomain = App::getEnv('_APP_DOMAIN_FUNCTIONS', 'disabled');
+        if ($functionsDomain !== 'disabled' && \str_ends_with($domain->get(), $functionsDomain)) {
+            $status = 'verified';
+        }
+
+        $rule = $dbForConsole->createDocument('rules', new Document([
+            '$id' => $ruleId,
+            'projectId' => $project->getId(),
+            'projectInternalId' => $project->getInternalId(),
+            'domain' => $domain->get(),
+            'resourceType' => $resourceType,
+            'resourceId' => $resourceId,
+            'resourceInternalId' => $resourceInternalId,
+            'status' => $status,
```

Comment:
Can"status" be a constant INT?

Reply:
We will refactor all this kind of ENUMs like fields as a separate PR.